api/src/opentrons/protocol_runner/legacy_command_mapper.py

```diff
@@ -276,6 +281,166 @@ def _build_initial_command(
 
         return engine_command
 
+    def _build_drop_tip_command(
+        self,
+        command: legacy_command_types.DropTipMessage,
+        command_id: str,
+        now: datetime,
+    ) -> pe_commands.Command:
+        well: LegacyWell
+        pipette: LegacyPipetteContext = command["payload"]["instrument"]
+        location = command["payload"].get("location")
+        assert isinstance(location, Location), "Unknown drop_tip location."
+        well = location.labware.as_well()
```

Comment:
- Why is this a `.get`; the field must be present according to the message's type definition?
- `location` is typed as a `Well` or `Location`; why are we asserting when we need a `Well`, anyway?

```suggestion
        location = command["payload"]["location"]
        well = (
            location if isinstance(location, LegacyWell) else location.labware.as_well()
        )
```